src/Microsoft.OData.Core/WriterValidator.cs

```diff
@@ -35,9 +54,17 @@ internal WriterValidator(ODataMessageWriterSettings settings)
         /// <returns>The created instance.</returns>
         public IDuplicatePropertyNameChecker CreateDuplicatePropertyNameChecker()
         {
-            return settings.ThrowOnDuplicatePropertyNames
-                   ? (IDuplicatePropertyNameChecker)new DuplicatePropertyNameChecker()
-                   : (IDuplicatePropertyNameChecker)new NullDuplicatePropertyNameChecker();
+            return this.duplicatePropertyNameCheckerObjectPool.GetObject();
+        }
+
+        /// <summary>
+        /// Release a DuplicatePropertyNameChecker instance.
+        /// </summary>
+        /// <returns>The created instance.</returns>
+        public void ReleaseDuplicatePropertyNameChecker(IDuplicatePropertyNameChecker duplicatePropertyNameChecker)

```

Comment:
Curious if it would make sense to use `IDisposable` pattern for this

Reply:
Making the `DuplicatePropertyNameChecker` disposable or the `ObjectPool` disposable?

Reply:
@habbes `DuplicateNameChecker` seeing as it's always calling "create" -> "do stuff" -> "release"

Reply:
Decided to keep the implementation as it is. We don't need to tightly couple the object pool and the `DuplicatePropertyNameChecker`.
Similar concept with the `ArrayPool` where we call `Rent` and `Return` directly.